integration/spark/src/main/scala/org/apache/carbondata/index/secondary/SecondaryIndexModel.java

```diff
@@ -99,5 +101,13 @@ public void setFetched(boolean fetched) {
     public Map<String, Set<String>> getSegmentToPosReferences() {
       return segmentToPosReferences;
     }
+
+    public void setValidSISegments(Set<String> validSISegments) {
```

Comment:
SecondaryIndexModel already has validSegment list. Please use that

Reply:
This segment list is nothing but the list of segments pruned by default index. Added validation to check if segment list present in SI and then used here.